docs/blog/content/material/database.en.md

```diff
@@ -4,4 +4,296 @@ weight = 6
+
+### 3 Core functions
+
+Data sharding, distributed transactions, elastic scaling, and distributed governance are the four core functions of Apache ShardingSphere at the current stage.
+
+#### Data sharding
+
+Divide and conquer is the solution used by Apache ShardingSphere to process massive data. Apache ShardingSphere enables distributed storage capabilities in databases through data sharding solutions.
+
+It can automatically route SQL to the corresponding data node according to the user's preset sharding algorithm to achieve the purpose of operating multiple databases. Users can use multiple databases managed by Apache ShardingSphere like a stand-alone database. Currently supports MySQL, PostgreSQL, Oracle, SQLServer and any database that supports SQL92 standard and JDBC standard protocol. The core flow of data sharding is shown in the figure below:
+
+![](https://shardingsphere.apache.org/blog/img/database6.jpg)
+
+The main process is as follows:
+
+1. Obtain the SQL and parameters input by the user by parsing the database protocol package or JDBC driver;
+    
+2. Parse SQL into AST (Abstract Syntax Tree) according to lexical analyzer and grammar analyzer, and extract the information required for fragmentation;
+    
+3. Match the shard key according to the user preset algorithm and calculate the routing path;
+    
+4. Rewrite SQL as distributed executable SQL;
+    
+5. Send SQL to each data node in parallel, the execution engine is responsible for balancing the connection pool and memory resources;
+    
+6. Perform streaming or full memory result set merge calculation according to AST;
+    
+7. Encapsulate the database protocol package or JDBC result set, and return to the client.
+    
+
+#### Distributed transaction
+
+Transaction is the core function of the database system. Distributed uncertainty and transaction complexity determine that there is no one-size-fits-all solution in the field of distributed transactions.
+
+Facing the current situation of blooming flowers, Apache ShardingSphere provides a highly open solution that uses standard interfaces to unify and integrate third-party distributed transaction frameworks independently selected by developers to meet the application requirements of various scenarios. In addition, Apache ShardingSphere also provides a new distributed transaction solution JDTX to make up for the lack of existing solutions.
+
+Standardized integrated interface
+
+Apache ShardingSphere provides a unified adaptation interface for local transactions, two-phase transactions, and flexible transactions, and docks with a large number of existing third-party mature solutions. Through the standard interface, developers can easily integrate other integration solutions into the Apache ShardingSphere platform.
```

Comment:
`flexible transactions` should be `BASE transactions`

Reply:
related also be replaced.